src/python/dxpy/workflow_builder.py

```diff
@@ -383,32 +418,47 @@ def _build_regular_workflow(json_spec, keep_open=False):
     return workflow_id
 
 
-def _get_validated_enabled_regions(json_spec, from_command_line):
+def _get_validated_enabled_regions(json_spec, args):
     """
     Returns a set of regions (region names) in which the global workflow
-    should be enabled. Also validates and synchronizes the regions
-    passed via CLI argument and in the regionalOptions field.
+    should be enabled. 
+    
+    1. validates and synchronizes the regions passed via CLI argument and in the regionalOptions field.
+    2. checks if these regions are included in the permitted regions of the bill_to
+    3. checks if the dependencies in all the stages are enabled in these regions
     """
+    # Determine in which regions the global workflow are requested to be available
     enabled_regions = dxpy.executable_builder.get_enabled_regions('globalworkflow',
                                                                   json_spec,
-                                                                  from_command_line,
+                                                                  args.region,
                                                                   WorkflowBuilderException)
     if not enabled_regions:
         enabled_regions = []
-        if not dxpy.WORKSPACE_ID:
+        if not dxpy.PROJECT_CONTEXT_ID:
```

Comment:
I changed it to project context id based on http://autodoc.dnanexus.com/bindings/python/current/dxpy.html since we are not running a job when building gwf. @commandlinegirl